### PR TITLE
Fix for #872 mstatus.GVA

### PR DESCRIPTION
**Pull request by rbuchner-aril**
Replaces xlate_flags with a data structure that is passed around and know the privilege of a load or store.

### CODE REVIEW

**Review by scottj97 — APPROVED** [review]
I've worked with @rbuchner-aril on this internally. Once the CI issues are fixed (do I need to update my GCC to find this? Or is this only a clang thing?) then I think this is good to go.

**Comment by aswaterman** [comment]
Thank you, @rbuchner-aril.  I'll review tomorrow.

@scottj97 It appears to fail only on the Ubuntu target and not the MacOS target, so it's likely that upgrading GCC is necessary to replicate the bug locally.  But it looks trivial:

```
/home/runner/work/riscv-isa-sim/riscv-isa-sim/ci-tests/../riscv/mmu.cc:63:8: error: unused variable ‘hlvx’ [-Werror=unused-variable]
   63 |   bool hlvx = access_info.flags.hlvx;
```

File: riscv/mmu.h [hunk 1]
```diff
@@ -38,6 +38,12 @@ struct tlb_entry_t {
   reg_t target_offset;
 };
 
+struct xlate_flags_t {
```
Comment (aswaterman) [inline riscv/mmu.h:41]: For reasons I didn't have time to thoroughly investigate, passing this struct by value reduces performance substantially on my Mac M1.

If you make it a bitfield struct (i.e. the below), the perf problem goes away while preserving the semantics.

```
struct xlate_flags_t {
  const bool forced_virt : 1;
  const bool hlvx : 1;
  const bool lr : 1;
};
```

**Review by aswaterman — CHANGES_REQUESTED** [review]
Looks great; once my minor performance comment is addressed, we should merge this.

File: riscv/mmu.h (hunk 1 above)
Comment (rbuchner-aril) [reply riscv/mmu.h:41]: Updated.

Comment (aswaterman) [reply riscv/mmu.h:41]: Thanks!

**Review by aswaterman — APPROVED** [review]
LGTM now.

**Pull request merged**